File: src/main/java/org/distril/beengine/player/data/provider/PlayerDataProvider.java
package org.distril.beengine.player.data.provider;

import org.distril.beengine.player.data.PlayerData;

import java.io.IOException;
import java.util.UUID;

public interface PlayerDataProvider {

	void save(UUID uuid, PlayerData data);

	PlayerData load(UUID uuid) throws IOException;
}

File: src/main/kotlin/org/distril/beengine/player/data/provider/PlayerDataProvider.kt
package org.distril.beengine.player.data.provider

import org.distril.beengine.player.data.PlayerData
import java.util.*

interface PlayerDataProvider {

	fun save(uuid: UUID, data: PlayerData, handler: (Throwable?) -> Unit)

	fun load(uuid: UUID, handler: (PlayerData, Throwable?) -> Unit)
}
